src/python/pants/bin/pants_runner.py

```diff
@@ -64,11 +64,13 @@ def _should_run_with_pantsd(self, global_bootstrap_options: OptionValueContainer
 
     @staticmethod
     def scrub_pythonpath() -> None:
-        # If PYTHONPATH was used to set up the Pants runtime environment, its entries are now on our
-        # `sys.path` allowing us to run. Do not propagate any of these Pants-specific sys.path entries
-        # forward to our subprocesses.
+        # Do not propagate any PYTHONPATH that happens to have been set in our environment
+        # to our subprocesses.
+        # Note that don't warn (but still scrub) if RUNNING_PANTS_FROM_SOURCES is set. This allows
+        # scripts that run pants directly from sources, and therefore must set PYTHONPATH, to mute
+        # this warning.
         pythonpath = os.environ.pop("PYTHONPATH", None)
-        if pythonpath:
+        if pythonpath and not os.environ.pop("RUNNING_PANTS_FROM_SOURCES", None):
```

Comment:
Can simplify:
```suggestion
        if pythonpath and "RUNNING_PANTS_FROM_SOURCES" not in os.environ:
```

Reply:
That does not have the same behavior. In the existing version I can turn the muting off by setting `RUNNING_PANTS_FROM_SOURCES=`, in your suggestion I couldn't turn it off except by calling `unset`. I had it that way for that reason.